### PR TITLE
Migrate pkg/kubelet/kuberuntime to structured logging

### PR DESCRIPTION
**Pull request by krzysiekg**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Migrate pkg/kubelet/kuberuntime to structured logging

#### Which issue(s) this PR fixes:

Ref #98976

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig node
/sig instrumentation
/priority important-soon

### CODE REVIEW

File: pkg/kubelet/kuberuntime/kuberuntime_container_linux.go [hunk 1]
```diff
@@ -112,13 +112,13 @@ func GetHugepageLimitsFromResources(resources v1.ResourceRequirements) []*runtim
...
 		if err != nil {
-			klog.Warningf("Failed to get hugepage size from resource name: %v", err)
+			klog.InfoS("Failed to get hugepage size from resource", "object", resourceObj, "err", err)
 			continue
 		}
 
 		sizeString, err := v1helper.HugePageUnitSizeFromByteSize(pageSize.Value())
 		if err != nil {
-			klog.Warningf("pageSize is invalid: %v", err)
+			klog.InfoS("Hugepage size is invalid", "object", resourceObj, "err", err)
```
Comment (shubheksha) [inline pkg/kubelet/kuberuntime/kuberuntime_container_linux.go:121]: I wouldn't add `Hugepage` here

File: pkg/kubelet/kuberuntime/kuberuntime_container_windows.go [hunk 2]
```diff
@@ -97,7 +97,7 @@ func (m *kubeGenericRuntimeManager) generateWindowsContainerConfig(container *v1
 	if wc.Resources.CpuCount > 0 {
 		if wc.Resources.CpuMaximum > 0 {
 			wc.Resources.CpuMaximum = 0
-			klog.Warningf("Mutually exclusive options: CPUCount priority > CPUMaximum priority on Windows Server Containers. CPUMaximum should be ignored")
+			klog.InfoS("Mutually exclusive options CPUCount and CPUMaximum present, CPUMaximum value ignored")
```
Comment (shubheksha) [inline pkg/kubelet/kuberuntime/kuberuntime_container_windows.go:100]: Why has the error message been modified here?

**Comment by shubheksha** [comment]
Hey @krzysiekg, we normally don't use full stops after error messages. Logging both the object and the specific info about the same object seems repetitive. I'd also advice not modifying the error messages too much unless to clarify them further. If you could address this across your PR, I'm happy to take another look.

File: pkg/kubelet/kuberuntime/kuberuntime_container_linux.go (hunk 1 above)
Comment (krzysiekg) [reply pkg/kubelet/kuberuntime/kuberuntime_container_linux.go]: Done

File: pkg/kubelet/kuberuntime/kuberuntime_container_windows.go (hunk 2 above)
Comment (krzysiekg) [reply pkg/kubelet/kuberuntime/kuberuntime_container_windows.go]: I was trying to simplify the message somehow, but I doesn't look good to me today. It's probably best to leave the original message intact. I reverted my changes.

**Review by kikisdeliveryservice — COMMENTED** [review]
Apologies wrong PR..

**Review by ehashman — COMMENTED** [review]
/lgtm
/milestone v1.21

**Comment by ehashman** [comment]
@mrunalp can you reapprove? the bot missed you

File: pkg/kubelet/kuberuntime/kuberuntime_sandbox.go [hunk 3]
```diff
@@ -38,15 +38,15 @@ func (m *kubeGenericRuntimeManager) createPodSandbox(pod *v1.Pod, attempt uint32
 	podSandboxConfig, err := m.generatePodSandboxConfig(pod, attempt)
 	if err != nil {
 		message := fmt.Sprintf("GeneratePodSandboxConfig for pod %q failed: %v", format.Pod(pod), err)
```
Comment (mrunalp) [inline pkg/kubelet/kuberuntime/kuberuntime_sandbox.go:40]: Update the message as well.

File: pkg/kubelet/kuberuntime/kuberuntime_sandbox.go [hunk 4]
```diff
@@ -38,15 +38,15 @@ func (m *kubeGenericRuntimeManager) createPodSandbox(pod *v1.Pod, attempt uint32
...
-		klog.Error(message)
+		klog.ErrorS(err, "Failed to generate sandbox config for pod", "pod", klog.KObj(pod))
 		return "", message, err
 	}
 
 	// Create pod logs directory
 	err = m.osInterface.MkdirAll(podSandboxConfig.LogDirectory, 0755)
 	if err != nil {
 		message := fmt.Sprintf("Create pod log directory for pod %q failed: %v", format.Pod(pod), err)
```
Comment (mrunalp) [inline pkg/kubelet/kuberuntime/kuberuntime_sandbox.go:48]: Update the message as well.

File: pkg/kubelet/kuberuntime/kuberuntime_sandbox.go [hunk 5]
```diff
@@ -58,14 +58,14 @@ func (m *kubeGenericRuntimeManager) createPodSandbox(pod *v1.Pod, attempt uint32
...
 		}
 		if runtimeHandler != "" {
-			klog.V(2).Infof("Running pod %s with RuntimeHandler %q", format.Pod(pod), runtimeHandler)
+			klog.V(2).InfoS("Running pod with runtime handler", "pod", klog.KObj(pod), "runtimeHandler", runtimeHandler)
 		}
 	}
 
 	podSandBoxID, err := m.runtimeService.RunPodSandbox(podSandboxConfig, runtimeHandler)
 	if err != nil {
 		message := fmt.Sprintf("CreatePodSandbox for pod %q failed: %v", format.Pod(pod), err)
```
Comment (mrunalp) [inline pkg/kubelet/kuberuntime/kuberuntime_sandbox.go:67]: Update the message as well.

File: pkg/kubelet/kuberuntime/kuberuntime_sandbox.go (hunk 3 above)
Comment (krzysiekg) [reply pkg/kubelet/kuberuntime/kuberuntime_sandbox.go]: I updated the messages. I also changed one more message that didn't have matching klog.ErrorS call to make all messages in this method consistent.

**Comment by ehashman** [comment]
Hi @krzysiekg, can you squash your commits? I think this is good to go otherwise

**Review by ehashman — COMMENTED** [review]
/lgtm

**Review by mrunalp — APPROVED** [review]

**Pull request merged**